### PR TITLE
add maximum expiration logic

### PR DESCRIPTION
**Pull request by bt-nia**
Disclaimer: I never programmed RUST and this is mostly LLM generated code

This aims to implement https://github.com/matze/wastebin/issues/54

cc: @matze @denissteinhorst for visibility

### CODE REVIEW

**Comment by bt-nia** [comment]
any update on this? :)

**Comment by matze** [comment]
I left a change request a while ago, regarding the `max_expiration()` function.

**Comment by bt-nia** [comment]
@matze can you please point me to it? I can't find it.

**Comment by matze** [comment]
Wait, you can't see the "Please return Option<i64> and Ok(None) here." referring to line 89 in` env.rs`?

**Comment by bt-nia** [comment]
Not sure where I can see that? This is my change view: 
image
I've never submitted change requests in Github, but I can say that in Gitlab it is easy to accidentally only submit comments for review but never actually _post_ the review.

**Review by matze — CHANGES_REQUESTED** [review]

**Comment by matze** [comment]
Sorry, you couldn't see it until I submitted the review :see_no_evil:

**Comment by bt-nia** [comment]
I was able to test the code now and made some improvements to the index.html template. 
I looked into your proposal, but from what it seems to me it is mostly a complication that makes value types less predictable. 

Unless you see any glaring issues, in which case: please let me know, could we please merge this? It is a feature we would really like to use :)

File: src/env.rs [hunk 1]
```diff
@@ -82,6 +83,14 @@ pub fn metadata() -> &'static Metadata<'static> {
     })
 }
 
+pub fn max_expiration() -> Result<i64, Error> {
+    std::env::var(VAR_MAX_EXPIRATION)
+        .map_or_else(
+            |_| Ok(-1), // Default to -1 if not set
```
[the remark this answers is not in the document]
Comment (matze) [reply src/env.rs:89]: > Given that Rust is a typed language, I would argue it is better to return a predictable type here instead.

But `Option<i64>` is a proper type and `None` reflects the absence of a value better than `-1`.

**Review by matze — CHANGES_REQUESTED** [review]

File: src/env.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (bt-nia) [reply src/env.rs]: I adjusted the code.

**Pull request closed without merging**